etl/src/main/java/org/pih/hivmigration/etl/sql/InfantMigrator.groovy

```diff
@@ -0,0 +1,142 @@
+package org.pih.hivmigration.etl.sql
+
+import org.w3c.dom.Text
+
+class InfantMigrator extends SqlMigrator {
+
+    @Override
+    def void migrate() {
+
+        executeMysql("Create Infants Staging Table", '''
+            create table hivmigration_infants (
+                person_id int PRIMARY KEY AUTO_INCREMENT,
+                source_infant_id int,
+                mother_patient_id int,
+                person_uuid char(38),
+                infant_code varchar(20),
+                first_name varchar(100),
+                last_name varchar(100),
+                gender varchar(50),
+                birthdate date
+            );
+        ''')
+
+        setAutoIncrement("hivmigration_infants", "(select (max(person_id)+1) from person)")
+
+        loadFromOracleToMySql('''
+                insert into hivmigration_infants (
+                    source_infant_id,
+                    mother_patient_id,
+                    infant_code,
+                    first_name,
+                    last_name,
+                    gender,
+                    birthdate
+                ) values (?, ?, ?, ?, ?, ?, ?)
+            ''', '''
+                select 
+                    i.INFANT_ID,
+                    i.MOTHER_PATIENT_ID,
+                    i.INFANT_CODE, 
+                    i.FIRST_NAME,
+                    i.LAST_NAME, 
+                    upper(i.GENDER), 
+                    i.BIRTH_DATE  			
+                from HIV_INFANTS i
+        ''')
+
+        executeMysql("Add UUIDs", "update hivmigration_infants set person_uuid = uuid();")
+
+        executeMysql("Load to person table", '''
+            insert into person
+              (person_id, uuid, gender, birthdate, creator, date_created)
+            select
+              i.person_id,
+              i.person_uuid,
+              COALESCE(i.gender, 'U'),  # most of the data is missing gender -- default to U
```

Comment:
I forget, does OpenMRS now support this? (U for gender?)

Reply:
No, but it doesn't break anything, while NULL does. It just shows "U" in the places you would expect a gender. There are even some message codes that you can define with it.

This is how I migrated missing genders for Mexico; no regrets yet!